### PR TITLE
create design system button stories

### PR DESCRIPTION
**Pull request by bhushankhope**

### CODE REVIEW

**Review by LibbyUX — COMMENTED** [review]
Hi, Bhushan! This looks great and it's almost ready. 

My only question:
- Is it possible to see the selected state for the nav cat button? It will need to hold the red underline in the selected state.

Thanks so much for everything!

image

**Comment by bhushankhope** [comment]
@LibbyUX Check now. Added underline on selected state. [Link](https://github.com/hubmapconsortium/hra-ui/pull/658#issuecomment-2299597337)

**Review by LibbyUX — APPROVED** [review]
Thanks so much @bhushankhope!

File: libs/design-system/button/src/lib/button.component.stories.ts
Comment (axdanbol) [inline libs/design-system/button/src/lib/button.component.stories.ts:1]: Please change all `<a>` tags to `<button>`. It is quite frustrating that it opens a new page every time you click a button. (Remove href and target attributes as well)

File: libs/design-system/button/src/lib/button-size/button-size.directive.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { computed, Directive, input } from '@angular/core';
+
+/** Input options for icon button size */
+export type ButtonSize = 'small' | 'medium' | 'large';
+
+/** Interface for button size and font variable */
+interface ButtonSizeAndFont {
```
Comment (axdanbol) [inline libs/design-system/button/src/lib/button-size/button-size.directive.ts:7]: Rename to `ButtonConfig`

File: libs/design-system/button/src/lib/button-size/button-size.directive.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  /** Size of the button */
+  size: number;
+  /** Font variable for the button */
+  font: string;
+  /** Padding for the current button size */
+  horizontalPadding: number;
+}
+
+/** Record of button sizes (number in rem) */
+const BUTTON_SIZES: Record<string, ButtonSizeAndFont> = {
```
Comment (axdanbol) [inline libs/design-system/button/src/lib/button-size/button-size.directive.ts:17]: Rename to `BUTTON_CONFIG` and update type to `Record<ButtonSize, ButtonConfig>`

**Review by axdanbol — CHANGES_REQUESTED** [review]
A couple of small tweaks

**Review by axdanbol — APPROVED** [review]

**Pull request merged**